Library/Homebrew/cmd/bottle.rb

```diff
@@ -130,7 +130,9 @@ def bottle_formula f
     else
       ohai "Determining #{f.name} bottle revision..."
       versions = FormulaVersions.new(f)
-      max = versions.bottle_version_map("origin/master")[f.pkg_version].max
+      bottle_revisions = versions.bottle_version_map("origin/master")[f.pkg_version]
+      pkg_reset = bottle_revisions[-2].zero? && !bottle_revisions[-1].zero?
+      max = pkg_reset ? bottle_revisions[0..-2].max : bottle_revisions.max
```

Comment:
`bottle_version_map` accurately records the bottle revisions throughout git history, so this just infers a bottle revision reset if the first bottle revision in history is > 0, and the subsequent one is 0 (i.e. the bot resets the revision from the previous pkg_version).

It's a bit confusing because the revisions are prepended onto an array such that the last item is the chronologically first. I'll try to make it more clear.